### ISSUE TITLE
[Linux] RPC not working 

### ISSUE THREAD

**Issue by AriTheFirst** [Vencord/Vesktop#481]
**Describe the bug**
Apps do not appear in my profile through Discord RPC. They do appear when using the regular discord client.

**To Reproduce**
1. Launch Vesktop
2. Launch Minecraft
3. Check the area on Vesktop where the "Currently Playing" should appear
4. Nothing is there

**Expected behavior**
Expected to see Minecraft Java in the currently playing area on discord.

**Screenshots**

**Desktop (please complete the following information):**
- OS/Distro: Arch Linux
- Desktop Environment (linux only): KDE
- OS Version: 6.6.16-1-LTS
- Vesktop Version: 1.5.1-1

**Command line output**
```
$ vesktop
[arRPC > ipc] listening at /run/user/1000/discord-ipc-0
[arRPC > websocket] listening on 6463
[arRPC > process] started
```
**Additional Information**
- The "Enable Rich Presence via arRPC" option is set to on in Vesktop settings
- The arRPC service definetly runs, it just dosen't function correctly as it appears
- Underneath "Activity Settings" there is no "Registered Games" setting as there is in the regular discord client

**Comment by Sqaaakoi** [comment]
arRPC only supports apps that explicitly set **rich** presence data. Minecraft does not support this unless you use a mod.

**Comment by Vendicated** [comment]
whats the "currently playing" area? vesktop just sets your presence, aka your profile activity. it doesn't display it above the user panel

are you using vesktop via flatpak?

**Comment by Sqaaakoi** [comment]
Hi, sorry about that earlier. I was (previously) unaware that this functionality existed in arRPC. After you read this, you will probably understand why I wasn't even aware this was a feature.

I have done some research, and I think I might be able to explain why Minecraft doesn't get registered through arRPC.
Looking at https://github.com/OpenAsar/arrpc/blob/main/src/process/index.js#L43 we see that it only strictly checks the [full process path on Linux](https://github.com/OpenAsar/arrpc/blob/main/src/process/native/linux.js#L5), which seems like it would never match in the majority of cases, regardless of which game, as the full command line likely isn't just the filename [like it would detect on Windows](https://github.com/OpenAsar/arrpc/blob/main/src/process/native/win32.js#L3).
Discord's detectable list, available at https://discord.com/api/v9/applications/detectable lists the path for Minecraft as `>java` on macOS (arRPC does not check this platform field).
The `>` at the start of the process name seems to me like Discord is using that as a "does the process path end in that name"

tl;dr: arRPC's process scanning is fundamentally broken on Linux, and it doesn't respect the format of Discord's detectable games list.

**Comment by AriTheFirst** [comment]
Ah, I guess i have to wait until arRPC gets fixed for Linux. Well, thank you anyway!

**Comment by PolisanTheEasyNick** [comment]
I think we can close it because [PR #75](https://github.com/OpenAsar/arrpc/pull/75) for arrpc is merged.

**Comment by KraXen72** [comment]
i'm having the same issue - using the vscode RPC extension, i get this error.
```
2024-06-22 17:56:06.405 [error] TypeError: Cannot read properties of null (reading 'write')
	at IPCTransport.send (/home/nex/.vscode/extensions/icrawl.discord-vscode-5.8.0/dist/extension.js:2:24881)
	at /home/nex/.vscode/extensions/icrawl.discord-vscode-5.8.0/dist/extension.js:2:24968
	at new Promise (<anonymous>)
	at IPCTransport.close (/home/nex/.vscode/extensions/icrawl.discord-vscode-5.8.0/dist/extension.js:2:24925)
	at RPCClient.destroy (/home/nex/.vscode/extensions/icrawl.discord-vscode-5.8.0/dist/extension.js:2:18626)
	at login (/home/nex/.vscode/extensions/icrawl.discord-vscode-5.8.0/dist/extension.js:2:149260)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```
i'm using vencord from flatpak, on kde plasma 6.1 fedora 40
arrpc runs:
```
nex@asus-nex:~$ flatpak run dev.vencord.Vesktop
Using NVIDIA on Wayland, disabling gpu sandbox
Passing the following arguments to Electron: --ozone-platform-hint=auto --disable-gpu-sandbox
[3:0622/175732.021043:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0622/175732.336983:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0622/175732.337090:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[arRPC > ipc] listening at /run/user/1000/discord-ipc-0
[arRPC > websocket] listening on 6463
[arRPC > process] started
```

**Comment by gaussandhisgun** [comment]
> i'm using vencord from flatpak

flatpak versions of discord require a [hack](https://wiki.archlinux.org/title/Discord#Enabling_rich_presence_on_Flatpak) to support RPCs. dont forget to change the package name to your own

**Comment by NoiceFrog** [comment]
hi, i am having the same issue on Arch Linux. Nothing gets detected. I have tried Lunar client, official Minecraft Launcher, Spotify, Xonotic, Steam games like Stardew Valley and Combat Master, VScode and Neovim (both with discord rpc plugins installed).
The original issue was posted on April 3rd, this is really annoying, could you please fix it?

**Comment by Chillsmeit** [comment]
I've tried using the following to no avail<br>
```
ln -sf {app/dev.vencord.Vesktop,$XDG_RUNTIME_DIR}/discord-ipc-0
```
to check the  symlink is correct I did `ls -l $XDG_RUNTIME_DIR/discord-ipc-0` and got: 
```
/run/user/1000/discord-ipc-0 -> app/io.github.verticalsync.sunroof/discord-ipc-0
```
So yeah, seems fine to.
When I run Vesktop in terminal it just listens and doesn't grab anything:
```
[arRPC > ipc] listening at /run/user/1000/discord-ipc-0
[arRPC > websocket] listening on 6463
[arRPC > process] started
```

**The only way I managed to get it working is to just not use the built-in arRPC...**
I had to setup my own arRPC server from [https://github.com/OpenAsar/arrpc](url) and then use the WebRichPresence (arRPC) Plugin.

**Comment by TheRealFame** [comment]
> The flatpak "hack" didn't work for me at all. Basically the built-in arRPC doesn't work for me probably since I'm using Vesktop flatpak I tried:
> 
> ```
> ln -sf {app/dev.vencord.Vesktop,$XDG_RUNTIME_DIR}/discord-ipc-0
> ```
> 
> to check the symlink is correct I did `ls -l $XDG_RUNTIME_DIR/discord-ipc-0` and got:
> 
> ```
> /run/user/1000/discord-ipc-0 -> app/io.github.verticalsync.sunroof/discord-ipc-0
> ```
> 
> So yeah, seems fine to me When I run Vesktop in terminal it just listens and doesn't grab anything:
> 
> ```
> [arRPC > ipc] listening at /run/user/1000/discord-ipc-0
> [arRPC > websocket] listening on 6463
> [arRPC > process] started
> ```
> 
> **The only way I managed to get it working is to just not use the built-in arRPC at all...** I had to setup my own arRPC server from https://github.com/OpenAsar/arrpc and then use the WebRichPresence (arRPC) Plugin.

![image](https://github.com/user-attachments/assets/6b9a585c-840e-4446-83c3-8ebdd05298ca)

I got flatsteal open, Trying to figure out what exactly the command was so i can get it working again. (The deb packages are broken for me, And the commands were talking about the offical discord and not vesktop's rpc.)

**Comment by retroDEG** [comment]
> I got flatsteal open, Trying to figure out what exactly the command was so i can get it working again. (The deb packages are broken for me, And the commands were talking about the offical discord and not vesktop's rpc.)

exactly why i had to stop using official discord "unable to satisfy dependencies" debs are just broken (and i mean even when it was working it was annoying having to download a new deb multiple times a day instead of it updating on its own)

**Comment by ptkato** [comment]
Is this supposed to be like this? Rich presence is working fine with arRPC, but the "registered games" entry in options is missing.

![image](https://github.com/user-attachments/assets/fcec25f9-0204-486a-bcb9-a64d69325ef8)

**Comment by Joomsy** [comment]
> Is this supposed to be like this? Rich presence is working fine with arRPC, but the "registered games" entry in options is missing.
> 
> ![image](https://private-user-images.githubusercontent.com/10677806/363975405-fcec25f9-0204-486a-bcb9-a64d69325ef8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxOTcyNTcsIm5iZiI6MTcyNzE5Njk1NywicGF0aCI6Ii8xMDY3NzgwNi8zNjM5NzU0MDUtZmNlYzI1ZjktMDIwNC00ODZhLWJjYjktYTY0ZDY5MzI1ZWY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI0VDE2NTU1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMDlmYTFiMmM3NWFkNmE4MmJlYWZmODUxOWZmNTM5MWM4MWI1ZmVkOGUyNTc1OWMxYmZiY2JkZjI4YzcxNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5OnvjOxWzEalvcNFc0BESawBHjvC3qdFV_bQn5Rc0f8)

Yes, it is supposed to be like this. Vesktop is a wrapper for Discord Web. It's basically just a Chromium instance loading Desktop Web in a browser without an address bar, or other form controls. Web doesn't give you the option for this since it isn't a desktop application. This is probably the only shortcoming I've found in Vesktop, because it prevents the ability to detect wine and Linux native games that don't have RPC. This mainly affects things like Lutris, and Steam gets an exception since Vesktop just works off of the "known title" database maintained by Discord themselves.

**Comment by nethershaw** [comment]
No one who landed here wants a wrapper for Discord Web.

Tell them what the difference between Vesktop and Vescord is, which is still stupidly obvious _when you find it_ but not at all clearly advertised in the places where you'd actually obtain either of them. The fact that _neither mentions the other_ in an "are you looking for x?" sort of way scans as the two projects being either weirdly disconnected clones of one another or somehow interchangeable, both of which are false. Someone ought to stick the same dang feature matrix on the front page for both of them comparing the two.

**TL;DR: anyone who is asking this question probably wants vencord, not vesktop, and doesn't know the difference.**
